Open Judge System/Web/OJS.Web/Areas/Administration/Views/ExamGroups/Index.cshtml

```diff
@@ -0,0 +1,176 @@
+          .Editable(editable =>
+          {
+              editable.Mode(GridEditMode.PopUp);
+              editable.Window(w => w.Title(Resource.Add_user));
+              editable.TemplateName("AddUserToExamGroup");
+              editable.DisplayDeleteConfirmation(Resource.Remove_user_prompt);
+          })
+          .Pageable(x => x.Refresh(true))
+          .Sortable(x => x.Enabled(true).AllowUnsort(false))
+          .Filterable(x => x.Enabled(true))
+          .DataSource(data =>
+          {
+              data
+                  .Ajax()
+                  .ServerOperation(true)
+                  .Model(model => model.Id(m => m.UserId))
+                  .Sort(sort => sort.Add(field => field.Username))
+                  .Create(create => create.Action("AddUserToExamGroup", ControllerName, new { id = "#= Id #" }))
+                  .Read(read => read.Action("UsersInExamGroup", ControllerName, new { id = "#= Id #" }))
+                  .Destroy(destroy => destroy.Action("RemoveUserFromExamGroup", ControllerName, new { id = "#= Id #" }))
+                  .Events(ev => ev.Error("validateModelStateErrors"));
+          })
+          .Events(ev => ev.DataBound("onDetailDataBound"))
+          .ToClientTemplate())
+</script>
+
+<script type="text/javascript">
+    function filterContests() {
+        return {
+            contestFilter: $('#ContestId').data("kendoComboBox").input.val()
+        }
+    }
+
+    function filterUsers() {
+        return {
+            userFilter: $('#UserId').data("kendoComboBox").input.val()
+        }
+    }
+
+    function onEdit() {
+        $('.k-edit-form-container').css('width', '100%');
+        $('.editor-label').css('width', '10%');
+        $('.editor-field').css('width', '80%');
+
+        var disabledFields = $('input:disabled').parent('.editor-field');
+
+        $.each(disabledFields, function (i, field) {
+            $(field).hide();
+            $(field).prev().hide();
+        });
+
+        $('#ContestId').change(function () {
+            $('#ContestName').val($('#ContestId_option_selected').text()).change();
+        });
+    }
+
+    function onDetailExpand(e) {
+        var contestName = $(e.masterRow).find('td:nth-child(4)').text();
+        if (contestName === '') {
+            var toolbarButtons = $(e.detailRow).find('.k-button');
+            disabableKendoButtons(toolbarButtons);
+
+            var message = $('#cannot-add-remove-users');
+            if (message.length) {
+                message.show();
+            } else {
+                toolbarButtons.last().after('<span id="cannot-add-remove-users">' +
+                    '<span class="glyphicon glyphicon-exclamation-sign"></span> @Resource.Cannot_manipulate_users</span>');
+            }
+        }
+    }
+
+    function onDetailDataBound() {
+        var detailsGrid = $('#' + this.element.attr("id"));
+        var contestName = detailsGrid.parents('tr').first().prev('.k-master-row').find('td:nth-child(4)').text();
```

Comment:
Is this a reliable way of grabbing an element? What if another child is added? Perhaps naming the element from the grid would be a more fool-proof approach